kedro/framework/cli/starters.py

```diff
@@ -149,11 +238,26 @@ def starter():
 @starter.command("list")
 def list_starters():
     """List all official project starters available."""
-    repo_url = _STARTERS_REPO.replace("git+", "").replace(".git", "/tree/main/{alias}")
-    output = [
-        {alias: repo_url.format(alias=alias)} for alias in sorted(_STARTER_ALIASES)
-    ]
-    click.echo(yaml.safe_dump(output))
+    starters_dict = _get_starters_dict()
+
+    # Group all specs by origin as nested dict and sort it.
+    sorted_starters_dict: Dict[str, Dict[str, KedroStarterSpec]] = {
+        origin: dict(sorted(starters_dict_by_origin))
+        for origin, starters_dict_by_origin in groupby(
+            starters_dict.items(), lambda item: item[1].origin
```

Comment:
Why is it `item[1]`? Could we just do `starters_dict.values()` and then `lambda value: value.origin`? I guess not because then you lose the keys?